stl_to_obj/stl_to_tscn.py

```diff
@@ -0,0 +1,259 @@
+
+
+def main():
+    # argparse to take in input.txt
+    parser = argparse.ArgumentParser(prog = ".stl to .obj and .tscn file converter")
+    parser.add_argument("input_file", type = str, help = "input json filename")
+    args = parser.parse_args()
+
+    # read in input .json file
+    with open(args.input_file, 'r') as f:
+        ######################
+        # .tscn file strings #
+        ######################
+        # all the .obj, .glb, .jpg (textures)
+        ext_resource = ""
+        ext_resource_id = 1
+
+        # SpatialMaterials (construction of textures)
+        sub_resource = ""
+        sub_resource_id = 1
+
+        # MeshInstances and collision scenes (.glb)
+        nodes = ""
+
+
+        ###################
+        # parse json file #
+        ###################
+        # load JSON object as dictionary
+        data = json.load(f)
+
+        header = data["header"]
+
+        input_folder = header["input_folder"]
+
+        # make the output folder
+        output_folder = header["output_folder"]
+        try:
+            os.mkdir(f"./{output_folder}")
+        # delete the folder and remake it if it does exist
+        except OSError as error:
+            print(f"NOTICE: {output_folder} folder already exists")
+            shutil.rmtree(output_folder) # delete folder and contents
+            os.mkdir(f"./{output_folder}")
+
+        # load provided textures
+        Textures.add_textures("textures.json")
+        # (optional) load extra textures 
+        if "extra_textures" in header: 
+            Textures.add_textures(header["extra_textures"])
+        
+        # initialize root node
+        nodes += f'[node name="{output_folder}" type="Spatial"]\n'
+
+        # (optional) set scale of scene
+        if "scale" in header:
+            scale = header["scale"]
+            nodes += f'transform = Transform( {scale}, 0, 0, 0, {scale}, 0, 0, 0, {scale}, 0, 0, 0 )\n'
+        nodes += "\n" # add extra new line to prepare for next node
+
+
+        ##################################
+        # iterate through all .stl files #
+        ##################################
+        meshes = data["meshes"]
+
+        # keep track of textures so we don't generate them more than once
+        texture_index = dict()
+
+        for m in meshes:
+            ####################
+            # generate texture #
+            ####################
+            texture = m["texture"]
+
+            # don't generate textures more than once
+            if texture not in texture_index:
+                ext_res_addon, ext_resource_id, sub_res_addon = generate_texture(texture, ext_resource_id, sub_resource_id)
+                ext_resource += ext_res_addon
+                sub_resource += f"{sub_res_addon}\n"
+                    
+                # add the sub resource's index to texture_index so we can use it later 
+                texture_index[texture] = sub_resource_id
+                sub_resource_id += 1 
```

Comment:
Do this in a function - with object oriented, this could be a member function with knowledge of the right data